Added abstract design patter

sweetashne opened the pull request.

sweetashne committed:
Added abstract design patter

Dpio commented on AbstractDesignPatter/BridgeOfSouls .cs:
```diff
@@ -0,0 +1,10 @@
+{
+	public class BridgeOfSouls : IBook
+	{
+		public string getName()
```
Please be consistent, the name of the functions should start with a capital letter.

sweetashne committed:
Function name changes.
"Please be consistent, the name of the functions should start with a capital letter." - Dpio
